packages/local_auth/lib/local_auth.dart

```diff
@@ -138,10 +138,19 @@ class LocalAuthentication {
 
   /// Returns true if device is capable of checking biometrics
   ///
+  /// Returns false on Android if device has the biometric
+  /// hardware but doesn't have any biometric enrolled
+  ///
   /// Returns a [Future] bool true or false:
-  Future<bool> get canCheckBiometrics async =>
-      (await _channel.invokeListMethod<String>('getAvailableBiometrics'))!
-          .isNotEmpty;
+  Future<bool> get canCheckBiometrics async {
+    final List<BiometricType> biometrics = await getAvailableBiometrics();
+    if (biometrics.isEmpty) {
+      return false;
+    } else if (_platform.isAndroid) {
+      return _canAuthenticateWithBiometrics;
```

Comment:
Why are you making a new Android-only method channel? The issue here is that `getAvailableBiometrics` is returning different conceptual information on Android and iOS; the fix should be in the implementation of that method.

Reply:
On iOS canCheckBiometrics returns false when device  doesn't have any fingerprint or face enrolled for Android canCheckBiometrics should work at the same way but currently this plugin returns true if device doesn't have any biometric enrolled because getAvailableBiometrics on Android return a list with the hardware biometrics available in device even if this doesn't have any fingerprint or face enrolled

Reply:
> @stuartmorgan are you saying that the change must be made in the java code

I'm saying that (under the current plugin design with a shared method channel), this should be fixed by making the Java behavior consistent with the Objective-C behavior.

> returning a empty list if the android device doesn't have any biometric enrolled ?

It appears that the current Obj-C behavior is to return `['undefined']` in that case, and that `canCheckBiometrics`'s implementation is based on that behavior.

> In that case What about devs who want to get the biometric hardware available on Android even if this doesn't have any fingerprint or face enrolled?

See above.